tests/include/detray/test/utils/monte_carlo.hpp

```diff
@@ -0,0 +1,84 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Detray tests include(s).
+#include "detray/test/types.hpp"
+
+// System include(s)
+#include <atomic>
+#include <random>
+#include <vector>
+
+namespace detray::test {
+
+/// This method creates a number of regular spaced points in 3D cartesian
+/// coordinates
+inline std::vector<point3> generate_regular_points(const scalar world,
+                                                   const std::size_t n_steps) {
+
+    const scalar d{world / static_cast<scalar>(n_steps)};
+
+    std::vector<point3> points{};
+    points.reserve(static_cast<std::size_t>(std::pow(n_steps, 3)));
+
+    for (std::size_t ix = 0u; ix < n_steps; ++ix) {
+        scalar x{-0.5f * world + static_cast<scalar>(ix) * d};
+
+        for (std::size_t iy = 0u; iy < n_steps; ++iy) {
+            scalar y{-0.5f * world + static_cast<scalar>(iy) * d};
+
+            for (std::size_t iz = 0u; iz < n_steps; ++iz) {
+                scalar z{-0.5f * world + static_cast<scalar>(iz) * d};
+
+                points.push_back({x, y, z});
+            }
+        }
+    }
+
+    return points;
+}
+
+/// This method creates a number of random points in 3D cartesian coordinates
+inline std::vector<point3> generate_random_points(
+    const scalar world, const std::size_t n_points,
+    const scalar seed = std::mt19937::default_seed) {
+    std::vector<point3> points{};
+    points.reserve(n_points);
+
+    std::random_device rd;
+    std::mt19937 mt(rd());
+    mt.seed(seed);
+    std::uniform_real_distribution<scalar> dist(-0.5f * world, 0.5f * world);
+
+    for (std::size_t i = 0u; i < n_points; ++i) {
+        points.push_back({dist(mt), dist(mt), dist(mt)});
+    }
+
+    return points;
+}
+
+/// Run a monte carlo experiment on the points in @param points
+template <typename predicate_t, typename... Args>
+inline scalar monte_carlo(const std::vector<point3> &points, Args... args) {
```

Comment:
Is this only for the `point3`? FYI, we already have a couple of monte-carlo tests.
What I can recall are landau_sampling and jacobian_validation. Probably more.

Reply:
That's all I needed so far, we can adapt it later if we need ```point2```, I guess